src/overloads/arrays.jl

```diff
@@ -114,6 +114,56 @@ for (Tx, TA, Ty) in Iterators.filter(
     @eval LinearAlgebra.dot(x::$Tx, A::$TA, y::$Ty) = LinearAlgebra.dot(x, A * y)
 end
 
+## Multiplication
+function Base.:*(A::Matrix{T}, B::Matrix) where {T<:AbstractTracer}
+    if size(A, 2) != size(B, 1)
+        throw(DimensionMismatch("arguments must have compatible dimensions"))
+    end
+    t = map(first_order_or, eachrow(A))
```

Comment:
Multiplication is twice differentiable in the cross-derivative, so this has to be `second-order-or`. 
There won't be a difference for `GradientTracer`, but there is no reason to use the wrong one here.

Reply:
The line you commented performs the equivalent of the sum of each row. So I don't think we need second order,

Reply:
What I'm saying is: I think this is correct even for Hessian tracing